### DISCUSSION TITLE
Private repository PR approval

### DISCUSSION THREAD

**Discussion by katelfox** [community/community#70261] [Pull Requests] [answered]
### Select Topic Area

Question

### Body

In a private repository can a user with read only access review/approve pull requests?

**Comment by andrej-kocijan** [comment, marked as answer]
I am going to assume you are talking about an organization owned repo since personal private repos can not have read-only collaborators. Read-only users can review pull request but they can not approve them or request changes. For more permissions details in organization owned repos see [this](https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/managing-repository-roles/repository-roles-for-an-organization).

If you were talking about an account owned repo, you can read about collaborators' permissions [here](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-user-account-settings/permission-levels-for-a-personal-account-repository)

**Reply by katelfox** [reply]
Hi Andrej,

Sorry yes I was talking about an organization owned repo.  I need to give access to a person to approve PRs but I do not want to give them write access to the repo.  

We don't have GitHub Enterprise, so I'm guessing I cannot do this.

Thank you for your very detailed answer!

**Comment by Dineth-H** [comment]
No, In a private GitHub repository, a **user with read-only access can only see the code and files**. They cannot review or approve pull requests, because this would allow them to make changes to the code that they do not have permission to make.

To review or approve pull requests, a user must have write access to the repository. You can give a user write access to the repository, or you can create a custom repository role with the permissions that they need.

Here are the different repository roles in GitHub:

- Read: Can see the code and files.
- Write: Can see, create, and edit the code and files.
- Maintain: Can see, create, edit, and delete the code and files. They can also manage the repository's settings and collaborators.
- Admin: Can see, create, edit, and delete the code and files. They can also manage the repository's settings and collaborators. They can also change the repository's ownership and delete the repository.

If you are not sure which repository role to give to a user, it is always best to err on the side of caution and give them the least amount of access necessary to perform their tasks.

I hope this is helpful!
